src/hwcap.c

```diff
@@ -14,13 +14,19 @@
 #ifdef HAVE_SYS_AUXV_H
 #	include <sys/auxv.h>
 #endif
+#ifdef __FreeBSD__
+#include <sys/sysctl.h>
+#else
 #include <sys/utsname.h>
-
-#ifndef __linux__
-#	error "Platform not supported (only Linux supported at the moment)"
 #endif
+
 #ifndef HAVE_GETAUXVAL
```

Comment:
```suggestion
#ifdef __FreeBSD__
```

FWICS `elf_aux_info` is specific to FreeBSD.

Reply:
I typically think that it's better to check for features rather than particular operating system or its version.  In this case, I found that `HAVE_GETAUXVAL` is exactly what I need: if no `getauxval()` provided by the system, fallback to `elf_aux_info()` as the only known (to me at this point) alternative.  But I won't insist, if you prefer `#ifdef __FreeBSD__` guard, it's okay, but then you should probably remove the check for `getauxval()` from the configure script.

Reply:
Yes, once this lands I plan to replace the build system with meson and I suppose I can remove the redundant check. I suppose another option would be to check for `getauxval` explicitly but TBH I don't think we can really expect that much portability between different platforms.